icicle/appUtils/ntt/ntt.cu

```diff
@@ -361,13 +361,13 @@ namespace ntt {
   template <typename S>
   class Domain
   {
-    static inline int max_size = 0;
-    static inline int max_log_size = 0;
-    static inline S* twiddles = nullptr;
-    static inline std::unordered_map<S, int> coset_index = {};
+    static inline std::unordered_map<int, int> max_size_for_device = {};
```

Comment:
I think it would be cleaner to make the fields non static and then have a static map from int to Domain.

Reply:
changed to array